features/support/sdapi.js

```diff
@@ -236,8 +236,49 @@ function cleanupToken(config) {
     });
 }
 
+/**
+ * abort all working builds
+ * @method cleanupBuilds
+ * @param  {Object}  config
+ * @param  {String}  config.instance      Screwdriver instance to test against
+ * @param  {String}  config.pipelineId    Pipeline ID to find the build in
+ * @param  {String}  config.jwt           JWT for authenticating
+ * @param  {String}  config.jobName       The job name we're looking for
+ * @return {Promise}
+ */
+function cleanupBuilds(config) {
+    const instance = config.instance;
+    const pipelineId = config.pipelineId;
+    const jwt = config.jwt;
+    const jobName = config.jobName;
+    const desiredStatus = ['RUNNING', 'QUEUED', 'BLOCKED', 'UNSTABLE'];
+
+    return findBuilds({
+        instance,
+        pipelineId,
+        jobName,
+        jwt
+    }).then((buildData) => {
+        const result = buildData.body || [];
+
+        result.filter(item => desiredStatus.includes(item.status)).forEach(build =>
```

Comment:
will this wait for all request promises to finish?

Reply:
I fixed it.